cmd/seid/cmd/iavl_parser.go

```diff
@@ -37,19 +49,86 @@ func BankParser(key []byte) ([]string, error) {
 		keyItems = append(keyItems, "Balances")
 		// remaining is length prefixed addr + denom
 		remaining := bytes.TrimPrefix(key, banktypes.BalancesPrefix)
-		lengthPrefix, remaining := int(remaining[0]), remaining[1:]
-		keyItems = append(keyItems, fmt.Sprintf("AddrLength: %d", lengthPrefix))
-		accountAddr := remaining[0:lengthPrefix]
-		denom := remaining[lengthPrefix:]
-		bech32Addr, err := sdk.Bech32ifyAddressBytes("sei", accountAddr)
+		items, denom, err := parseLengthPrefixedAddress(remaining)
 		if err != nil {
 			return keyItems, err
 		}
-		keyItems = append(keyItems, fmt.Sprintf("AddrBech32: %s", bech32Addr))
+		keyItems = append(keyItems, items...)
 		keyItems = append(keyItems, fmt.Sprintf("Denom: %s", string(denom)))
 	default:
 		keyItems = append(keyItems, "Unrecognized prefix")
 	}
 
 	return keyItems, nil
 }
+
+func DexParser(key []byte) ([]string, error) {
+	keyItems := []string{}
+	if matched, items, _, err := MatchAndExtractDexAddressPrefixKeys(key); matched {
+		if err != nil {
+			return keyItems, err
+		}
+		keyItems = append(keyItems, items...)
+		return keyItems, nil
+	}
+	switch {
+	case bytes.HasPrefix(key, []byte(dexkeeper.EpochKey)):
+		// do nothing since the key is a string and no other data to be parsed
+	default:
+		keyItems = append(keyItems, "Unrecognized prefix")
+	}
+	return keyItems, nil
+}
+
+func MatchAndExtractDexAddressPrefixKeys(key []byte) (bool, []string, []byte, error) {
+	keyItems := []string{}
+	keysToMatch := []string{
```

Comment:
might be helpful to add a source of truth can be found if it can't be updated automatically

Reply:
Added comment discussing source of truth. The constants are all in the dex keys file